packages/storage/src/reference.ts

```diff
@@ -145,19 +180,32 @@ export function uploadBytesResumable(
  * @returns An UploadTask that lets you control and
  *     observe the upload.
  */
-export function uploadString(
+export async function uploadString(
```

Comment:
We shouldn't make public API functions  `async` as it may alter their timing and exception behavior.

Reply:
Removed (shared code between uploadString and uploadBytes should be moved out to a shared function anyway I think).